monitoring/grafana/WDSAPIServerMonitoring-1722629699305.json

```diff
@@ -0,0 +1,686 @@
+          "instant": false,
+          "legendFormat": "__auto",
+          "range": true,
+          "refId": "A",
+          "useBackend": false
+        }
+      ],
+      "title": "etcd_request_duration_seconds_bucket",
+      "type": "timeseries"
+    },
+    {
+      "datasource": {
+        "default": true,
+        "type": "prometheus",
+        "uid": "prometheus-test"
+      },
+      "fieldConfig": {
+        "defaults": {
+          "color": {
+            "mode": "palette-classic"
+          },
+          "custom": {
+            "axisBorderShow": false,
+            "axisCenteredZero": false,
+            "axisColorMode": "text",
+            "axisLabel": "",
+            "axisPlacement": "auto",
+            "barAlignment": 0,
+            "barWidthFactor": 0.6,
+            "drawStyle": "line",
+            "fillOpacity": 0,
+            "gradientMode": "none",
+            "hideFrom": {
+              "legend": false,
+              "tooltip": false,
+              "viz": false
+            },
+            "insertNulls": false,
+            "lineInterpolation": "linear",
+            "lineWidth": 1,
+            "pointSize": 5,
+            "scaleDistribution": {
+              "type": "linear"
+            },
+            "showPoints": "auto",
+            "spanNulls": false,
+            "stacking": {
+              "group": "A",
+              "mode": "none"
+            },
+            "thresholdsStyle": {
+              "mode": "off"
+            }
+          },
+          "mappings": [],
+          "thresholds": {
+            "mode": "absolute",
+            "steps": [
+              {
+                "color": "green",
+                "value": null
+              },
+              {
+                "color": "red",
+                "value": 80
+              }
+            ]
+          }
+        },
+        "overrides": []
+      },
+      "gridPos": {
+        "h": 8,
+        "w": 12,
+        "x": 0,
+        "y": 16
+      },
+      "id": 5,
+      "options": {
+        "legend": {
+          "calcs": [],
+          "displayMode": "list",
+          "placement": "bottom",
+          "showLegend": true
+        },
+        "tooltip": {
+          "mode": "single",
+          "sort": "none"
+        }
+      },
+      "targets": [
+        {
+          "datasource": {
+            "type": "prometheus",
+            "uid": "prometheus-test"
+          },
+          "disableTextWrap": false,
+          "editorMode": "builder",
+          "expr": "histogram_quantile(0.99, sum by(le, service, verb) (rate(rest_client_request_duration_seconds_bucket{job=~\"$SpaceName\", namespace=\"$SpaceNS\"}[1m])))",
+          "fullMetaSearch": false,
+          "includeNullMetadata": true,
+          "instant": false,
+          "legendFormat": "__auto",
+          "range": true,
+          "refId": "A",
+          "useBackend": false
+        }
+      ],
+      "title": "rest_client_request_duration_seconds_bucket",
```

Comment:
A better title would be something like "99th percentile of duration of requests issued by apiserver"

Reply:
rest_client_request_duration_seconds_bucket measures the latency or duration in seconds for calls to the API server **as seen by the API server**. Is it?  See the extracts that I collected from an apiserver metrics endpoint: 

```
rest_client_request_duration_seconds_bucket{host="[::1]:6443",verb="GET",le="0.005"} 62737
rest_client_request_duration_seconds_bucket{host="[::1]:6443",verb="GET",le="0.025"} 62779
rest_client_request_duration_seconds_bucket{host="ingress-nginx-controller-admission.ingress-nginx.svc:443",verb="POST",le="0.005"} 0
rest_client_request_duration_seconds_bucket{host="ingress-nginx-controller-admission.ingress-nginx.svc:443",verb="POST",le="0.025"} 2
est_client_request_duration_seconds_bucket{host="prometheus-kube-prometheus-operator.ks-monitoring.svc:443",verb="POST",le="1"} 0
rest_client_request_duration_seconds_bucket{host="prometheus-kube-prometheus-operator.ks-monitoring.svc:443",verb="POST",le="2"} 70
rest_client_request_duration_seconds_bucket{host="prometheus-kube-prometheus-operator.ks-monitoring.svc:443",verb="POST",le="4"} 70
```

Reply:
Fixed! Replaced by a new query instead